packages/amplify-e2e-tests/src/codegen/add.ts

```diff
@@ -1,31 +1,44 @@
 import { nspawn as spawn, getCLIPath } from 'amplify-e2e-core';
 
-export function addCodegen(cwd: string, settings: any): Promise<void> {
-  return new Promise((resolve, reject) => {
-    const run = spawn(getCLIPath(), ['codegen', 'add'], { cwd, stripColors: true });
-    if (!(settings.ios || settings.android)) {
-      run.wait('Choose the code generation language target').sendCarriageReturn();
-    }
+type AddCodegenSettings = {
+  ios?: boolean;
+  android?: boolean;
+  apiId?: string;
+};
```

Comment:
Most of this file is just style changes, I added this type for the input, and added lines 15-18 to support adding codegen with an external api id.